packages/fast-components-styles-msft/src/badge/index.ts

```diff
@@ -0,0 +1,108 @@
+import {
+    ComponentStyles,
+    ComponentStyleSheet,
+    CSSRules,
+} from "@microsoft/fast-jss-manager";
+import { BadgeClassNameContract } from "@microsoft/fast-components-class-name-contracts-msft";
+import {
+    applyLocalizedProperty,
+    Direction,
+    ellipsis,
+    toPx,
+} from "@microsoft/fast-jss-utilities";
+import { DesignSystem, withDesignSystemDefaults } from "../design-system/index";
+import { ensureNormalContrast } from "../utilities/colors";
+import { fontWeight } from "../utilities/fonts";
+
+function smallBadgeStyle(direction: Direction): CSSRules<DesignSystem> {
+    return {
+        padding: direction === Direction.ltr ? "0 7px 2px 0" : "0 0 2px 7px",
+        height: "14px",
+        lineHeight: "16px",
+        "&$badge__highlight, &$badge__lowlight, &$badge__accent": {
+            padding: "0 7px 4px",
+        },
+    };
+}
+
+function largeBadgeStyle(direction: Direction): CSSRules<DesignSystem> {
+    return {
+        [applyLocalizedProperty("paddingRight", "paddingLeft", direction)]: "14px",
+        height: "20px",
+        "&$badge__highlight, &$badge__lowlight, &$badge__accent": {
+            padding: "2px 14px 4px",
+        },
+    };
+}
+
+function backplateStyle(designSystem: DesignSystem): CSSRules<DesignSystem> {
+    return {
+        borderRadius: toPx(designSystem.cornerRadius),
+    };
+}
+
+const styles: ComponentStyles<BadgeClassNameContract, DesignSystem> = (
+    config: DesignSystem
+): ComponentStyleSheet<BadgeClassNameContract, DesignSystem> => {
+    const designSystem: DesignSystem = withDesignSystemDefaults(config);
+    const direction: Direction = designSystem.direction;
+    const contrast: number = designSystem.contrast;
+    // Badges do not switch color on theme change
+    const foregroundColor: string = "#111";
+    const backgroundColor: string = "#FFF";
+    const lowlightBackground: string = "#333";
+    const highlightBackground: string = "#FFD800";
+    const accentBackground: string = designSystem.brandColor;
+    const accentForegroundColor: string = ensureNormalContrast(
+        contrast,
+        foregroundColor,
+        accentBackground
+    );
+    const hightlightForegroundColor: string = ensureNormalContrast(
+        contrast,
+        foregroundColor,
+        highlightBackground
+    );
+    const lowlightForegroundColor: string = ensureNormalContrast(
+        contrast,
+        backgroundColor,
+        lowlightBackground
+    );
+
+    return {
+        badge: {
+            fontSize: "13px",
+            lineHeight: "20px",
```

Comment:
everything should be on the new type ramp.

Reply:
14 or 12; I would ask @Falkicon, but this would create issues by being the only thing off the typeramp. It would be one thing if the delta was a line-height only; but the font being different and off doesn't really seem to make sense aside from the fact that we're porting this over from a previous implementation. Everything added should really be considered net-new work.

Reply:
I also noticed that this is different than "smallBadge" above.

Reply:
Have changed to `14px`. The small version's height had to be updated from `18px` to `20px` There is just not enough room inside to include the text (especially since it won't be all caps now).

Reply:
I've updated the docs to make the sizes more apparent, but for large the text should be 14/20 with a padding of 3px 12px, with an overall height of 26px. Small should be 12/16 with a padding of 2px 8px, overall height of 20px. The letter spacing should all be default.